### PR TITLE
Literal floating point Narrowing Conversions

### PR DESCRIPTION
**Pull request by cugone**
Closes #155.

Last part of splitting #155 up into smaller commits.

Commit this one last.

### CODE REVIEW

File: src/UI/Core/TextField.cpp [hunk 1]
```diff
@@ -139,13 +139,13 @@ void TextField::onTextInput(const std::string& _s)
 
 	if (mMaxCharacters > 0 && text().length() == mMaxCharacters) { return; }
 
-	int prvLen = text().length();
+	std::size_t prvLen = text().length();
```
Comment (DanRStevens) [inline src/UI/Core/TextField.cpp]: This change appears to be included in #173. It may be more appropriate in this PR.

File: src/States/MapViewStateDraw.cpp [hunk 2]
```diff
@@ -251,17 +247,15 @@ void MapViewState::drawNavInfo()
 	if (isPointInRect(MOUSE_COORDS, MOVE_DOWN_ICON))
 	{
 		r.drawSubImage(mUiIcons, MOVE_DOWN_ICON.x(), MOVE_DOWN_ICON.y(), 64, 128, 32, 32, 255, 0, 0, 255);
-	}
-	else
+	} else
```
Comment (DanRStevens) [inline src/States/MapViewStateDraw.cpp]: There's quite a few style changes like this, which should really be done in their own commit.

----

Not too sure how Leeor feels about the style used here. I prefer a more compact style, though I'm uncertain what the code base tends to use.

File: src/States/MapViewStateDraw.cpp [hunk 3]
```diff
@@ -151,57 +157,47 @@ void MapViewState::drawResourceInfo()
...
 
 	r.drawSubImage(mUiIcons, (x + offsetX) * 10, y, 176 + (std::clamp(mCurrentMorale, 1, 999) / 200) * constants::RESOURCE_ICON_SIZE, 0, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE);
 	r.drawText(*MAIN_FONT, string_format("%i", mPopulation.size()), (x + offsetX) * 10 + margin, textY, 255, 255, 255);
 
-	if (mPinPopulationPanel	|| isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 675, 1, 75, 19)) { mPopulationPanel.update(); }
-	if (mPinResourcePanel	|| isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 0, 1, mResourceBreakdownPanel.width(), 19)) { mResourceBreakdownPanel.update(); }
+	if (mPinPopulationPanel || isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 675, 1, 75, 19)) { mPopulationPanel.update(); }
+	if (mPinResourcePanel || isPointInRect(MOUSE_COORDS.x(), MOUSE_COORDS.y(), 0, 1, mResourceBreakdownPanel.width(), 19)) { mResourceBreakdownPanel.update(); }
```
Comment (DanRStevens) [inline src/States/MapViewStateDraw.cpp]: More style changes that should be in their own commit.

File: src/States/MapViewStateDraw.cpp [hunk 4]
```diff
@@ -151,57 +157,47 @@ void MapViewState::drawResourceInfo()
...
 
 	// Population / Morale
-	if (mCurrentMorale > mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 16, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); }
-	else if (mCurrentMorale < mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 0, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); }
-	else { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 32, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); }
+	if (mCurrentMorale > mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 16, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); } else if (mCurrentMorale < mPreviousMorale) { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 0, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); } else { r.drawSubImage(mUiIcons, (x + offsetX) * 10 - 17, y, 32, 48, constants::RESOURCE_ICON_SIZE, constants::RESOURCE_ICON_SIZE); }
```
Comment (DanRStevens) [inline src/States/MapViewStateDraw.cpp]: Excessively long line. Was this intentional? Other lines in this section look similar.

File: src/States/MapViewStateDraw.cpp [hunk 5]
```diff
@@ -69,8 +69,14 @@ void MapViewState::drawMiniMap()
...
-	if (mBtnToggleHeightmap.toggled()) { r.drawImage(mHeightMap, mMiniMapBoundingBox.x(), mMiniMapBoundingBox.y()); }
-	else { r.drawImage(mMapDisplay, mMiniMapBoundingBox.x(), mMiniMapBoundingBox.y()); }
+	if (mBtnToggleHeightmap.toggled())
+	{
+		r.drawImage(mHeightMap, mMiniMapBoundingBox.x(), mMiniMapBoundingBox.y());
+	}
+	else
+	{
+		r.drawImage(mMapDisplay, mMiniMapBoundingBox.x(), mMiniMapBoundingBox.y());
+	}
```
Comment (DanRStevens) [inline src/States/MapViewStateDraw.cpp]: Reformatting change, which should be in its own commit.

File: src/States/MainReportsUiState.cpp [hunk 6]
```diff
@@ -258,7 +257,7 @@ void MainReportsUiState::_deactivate()
 void MainReportsUiState::onKeyDown(EventHandler::KeyCode key, EventHandler::KeyModifier mod, bool repeat)
 {
 	if (!active()) { return; }
-	if(key == NAS2D::EventHandler::KeyCode::KEY_ESCAPE) { exit(); }
+	if (key == NAS2D::EventHandler::KeyCode::KEY_ESCAPE) { exit(); }
```
Comment (DanRStevens) [inline src/States/MainReportsUiState.cpp]: Reformatting change.

**Comment by cugone** [comment]
I apologize for the style and formatting changes. I was making undo commits to split up the `static_cast`s/literals and I missed a few `clang-format` auto-changes.

**Comment by cugone** [comment]
Style and formatting changes to `MapViewStateDraw.cpp` have been moved to PR #177.

Reverted the formatting changes to `MainReportsUiState.cpp` and moved them to PR #178.

File: src/UI/Core/TextField.cpp (hunk 1 above)
Comment (cugone) [reply src/UI/Core/TextField.cpp]: Done

**Comment by DanRStevens** [comment]
Thanks for looking into that. Sounds like the Clang auto formatting can be quite a pain.

There are merge conflicts now, which will need to be resolved. Might have been from the last branch I merged.

**Comment by cugone** [comment]
Resolved.

File: src/States/MapViewStateDraw.cpp (hunk 4 above)
Comment (DanRStevens) [reply src/States/MapViewStateDraw.cpp]: This has definitely not been resolved. The `if` and `else` blocks shouldn't all be on one long line like this.

**Comment by DanRStevens** [comment]
I think this one needs a bit of rework and attention. There is too much going on in a single commit.

File: src/States/MapViewStateDraw.cpp (hunk 4 above)
Comment (cugone) [reply src/States/MapViewStateDraw.cpp]: You are reviewing the wrong commit. Any change to MapViewStateDraw is part of #177. These issues have been resolved in that PR. Which is why I marked this resolved.

Comment (DanRStevens) [reply src/States/MapViewStateDraw.cpp]: That change is on this branch. If you don't want it reviewed here, it should be removed.

**Comment by DanRStevens** [comment]
New merge conflicts for this branch. The code formatting still needs to be updated as well.

**Comment by cugone** [comment]
Due to pixel-alignment requirements noted in other PRs, closing this to rework.
